moved compiler configuration into tsconfig.json

tomlarkworthy opened the pull request:
... so other tooling is configuration aware (e.g. vscode)
@mckoss

mckoss commented on gulpfile.js:
```diff
@@ -61,14 +61,6 @@ var TEST_SETS = [
 
-var TS_SETTINGS = {
-  sortOutput: true,
-  declarationFiles: true,
```
Why did we get rid of settings for sortOutput, declarationFiles, and noExternalResolve?

tomlarkworthy replied:
ASFAIK we don't save the .d.ts files so we don't need declarationFiles to be generated. 
The others are not actual compiler options.
sortFiles is used to preserve source-file maps across browserfy but it seems to works without this flag.
I am not sure why we had noExternalResolve becuase it also all seems to work without it. Do you know why we have them?

mckoss commented on tsconfig.json:
```diff
@@ -0,0 +1,32 @@
+    "noEmitOnError": true
+  },
+  "files": [
+	"src/ast.ts",
```
Wierd indentation?

mckoss commented on tsconfig.json:
```diff
@@ -0,0 +1,32 @@
+  "files": [
+	"src/ast.ts",
+	"src/bolt.ts",
+	"src/bolt.ts",
```
bolt.ts listed twice?

mckoss commented on tsconfig.json:
```diff
@@ -0,0 +1,32 @@
+    "sourceMap": true,
+    "noEmitOnError": true
+  },
+  "files": [
```
Can we not compile all *.ts files in src?

mckoss replied:
and src/test?

tomlarkworthy replied:
:) glob support in tsconfig is on the 1.8 roadmap!

mckoss commented:
NOTE: Our Travis CI server is reporting success - BUT - there are 0 tests being executed - don't merge until fixed.

tomlarkworthy commented:
Yeah I fixed the issue and run-tests works properly now. it was generated lib/test/test for due to the TS output dir being counted twice.. I removed the compile-tests step altogether as things were getting compiled twice and we never really make a downstream distinction in the workflow anyway. It runs fast this way (and it simplifies the tsconfig with one compile set). 

PTAL @mckoss
